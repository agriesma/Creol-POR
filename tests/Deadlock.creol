/*
 * Deadlock.creol
 *
 * Instances of this class may display a deadlock.
 *
 * main("Deadlock", emp)
 */
class Deadlock
begin
  op init == skip
  op run == var l: Label[ ]; l!n(); release; skip; l?()
  op n == release
end
