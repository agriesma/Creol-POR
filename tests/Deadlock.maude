load interpreter
mod PROGRAM is
pr CREOL-INTERPRETER .
op init : -> Configuration [ctor] .
eq init =
< "Deadlock" : Cl | Inh: noInh, Par: noAid, Att: noSubst, Mtds: 
  < "init" : Mtdname | Param: noAid, Latt: noSubst, Code: skip ; return ( emp ) > *
  < "run" : Mtdname | Param: noAid, Latt: noSubst, Code: "l" ! "this" . "n" ( emp ) ; release ; skip ; ( "l" ? ( noAid ) )  ; return ( emp ) > *
  < "n" : Mtdname | Param: noAid, Latt: noSubst, Code: release ; return ( emp ) >, Ocnt: 0 >

.
endm
