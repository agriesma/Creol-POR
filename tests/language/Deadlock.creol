/*
 * Deadlock.creol -- Corner cases of the Creol semantics.
 *
 * This file is part of creoltools
 *
 * Written and Copyright (c) 2007 by Marcel Kyas
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

/*
 * Instances of this class may display a deadlock.
 *
 * main("Deadlock", emp)
 */
class Deadlock
begin
  op init == skip
  op run == var l: Label[ ]; l!n(); release; skip; l?()
  op n == release
end
