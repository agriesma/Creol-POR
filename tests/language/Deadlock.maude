load creol-interpreter .
mod PROGRAM is
protecting CREOL-SIMULATOR .
op classes : -> Configuration [ctor] .
eq classes =
< "Deadlock" : Cl | Inh: (noInh),
  Par: (noVid),
  Att: (noSubst),
  Mtds: (< "init" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return ( emp ) > *
    < "run" : Mtdname | Param: noVid, Latt: "l" |-> null , "_" |-> null, Code: ( "l" ! "this" . "n" (emp ) )  ; release ; skip ; ( "l" ? ( noVid ) )  ; return ( emp ) > *
    < "n" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: release ; return ( emp ) >),
  Ocnt: 0 >

.
endm
