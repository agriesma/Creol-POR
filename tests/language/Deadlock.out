==========================================
reduce in PROGRAM : init .
result Class: 
< "Deadlock" : Cl | Inh: noInh, Par: noVid, Att: noSubst, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: skip ; return(
    emp) > *
  < "n" : Mtdname | Param: noVid, Latt: "_" |-> null, Code: release ; return(
    emp) > *
  < "run" : Mtdname | Param: noVid, Latt: "_" |-> null, "l" |-> null, Code: "l"
    ! "this" . "n"(emp) ; release ; skip ; "l" ?(noVid) ; return(emp) >,
  Ocnt: 0 >
Bye.
